apps/desktop/desktop_native/src/processisolation/linux.rs

```diff
@@ -0,0 +1,45 @@
+use anyhow::Result;
+use libc;
+
+pub fn disable_coredumps() -> Result<String> {
+    // RLIMIT_CORE is the maximum size of a core dump file. Setting both to 0 disables core dumps, on crashes
+    // https://github.com/torvalds/linux/blob/1613e604df0cd359cf2a7fbd9be7a0bcfacfabd0/include/uapi/asm-generic/resource.h#L20
+    const RLIMIT_CORE: u32 = 4;
+    let rlimit = libc::rlimit {
+        rlim_cur: 0,
+        rlim_max: 0,
+    };
+    if unsafe { libc::setrlimit(RLIMIT_CORE, &rlimit) } != 0 {
+        let e = std::io::Error::last_os_error();
+        return Err(anyhow::anyhow!("failed to disable core dumping, memory might be persisted to disk on crashes {}", e))
+    }
+
+    Ok("Core dumps disabled".to_string())
+}
```

Comment:
I think an empty result is just fine, here

```suggestion
pub fn disable_coredumps() -> Result<()> {
    // RLIMIT_CORE is the maximum size of a core dump file. Setting both to 0 disables core dumps, on crashes
    // https://github.com/torvalds/linux/blob/1613e604df0cd359cf2a7fbd9be7a0bcfacfabd0/include/uapi/asm-generic/resource.h#L20
    const RLIMIT_CORE: u32 = 4;
    let rlimit = libc::rlimit {
        rlim_cur: 0,
        rlim_max: 0,
    };
    if unsafe { libc::setrlimit(RLIMIT_CORE, &rlimit) } != 0 {
        let e = std::io::Error::last_os_error();
        return Err(anyhow::anyhow!("failed to disable core dumping, memory might be persisted to disk on crashes {}", e))
    }

    Ok(())
}
```